src/in-view.js

```diff
@@ -58,69 +66,76 @@ const inView = () => {
+        } else {
+            if (selector instanceof window.Node) {
+                elements = [selector];
+            } else if (selector instanceof window.NodeList || selector instanceof Array) {
+                elements = [].slice.call(selector);
+            } else {
+                // selector is not supported
+                return null;
+            }
+            // create unique selector
+            selector = 'Node-'+nodeCounter++;
+        }
 
         // If the registry exists, update the elements.
         if (selectors.history.indexOf(selector) > -1) {
-            selectors[selector].elements = elements;
+            // get selector
+            let sel = selectors[selector];
+            // validate new options
+            let o = sel.validate(opts);
+            // check that the options have not changed
+            if (isEqual(o, sel.options)) {
+                if (selector.substr(0,4) === 'Node') {
+                    sel.elements.concat(elements);
+                } else {
+                    sel.elements = elements;
+                }
+
+                return sel;
+            } else {
+                // if options have changed, modify selector and add it as new
+                // based on the selector
+                let count = selectorCounter[selector] = ~~selectorCounter[selector]+1;
+                selector += '-'+count;
+            }
         }
 
         // If it doesn't exist, create a new registry.
-        else {
-            selectors[selector] = Registry(elements, options);
-            selectors.history.push(selector);
-        }
+        selectors[selector] = Registry(elements, opts, selector);
 
-        return selectors[selector];
-    };
+        selectors.history.push(selector);
 
-    /**
-    * Mutate the offset object with either an object
-    * or a number.
-    */
-    control.offset = o => {
-        if (o === undefined) return options.offset;
-        const isNum = n => typeof n === 'number';
-        ['top', 'right', 'bottom', 'left']
-            .forEach(isNum(o) ?
-                dim => options.offset[dim] = o :
-                dim => isNum(o[dim]) ? options.offset[dim] = o[dim] : null
-            );
-        return options.offset;
+        return selectors[selector];
     };
 
     /**
-    * Set the threshold with a number.
-    */
-    control.threshold = n => {
-        return typeof n === 'number' && n >= 0 && n <= 1
-            ? options.threshold = n
-            : options.threshold;
-    };
+     * Attempts to get a registry from the selectors object
+     */
+    control.get = s => {
+        return selectors[s];
+    }
 
     /**
-    * Use a custom test, overriding inViewport, to
-    * determine element visibility.
-    */
-    control.test = fn => {
-        return typeof fn === 'function'
-            ? options.test = fn
-            : options.test;
-    };
+     * Add proxy to global check
+     */
+    control.check = () => check();
```

Comment:
wouldn't it be better to add `check` to Registries too, so that they only check their own selectors?

Reply:
I'm not following. There's already a public `check` in the registry that you can call directly `inview(".foo").check()`